server/src/test/java/io/crate/analyze/expressions/ExpressionAnalyzerTest.java

```diff
@@ -387,6 +388,17 @@ public void test_resolve_eq_function_for_text_types_with_length_limit_and_unboun
             s -> assertThat(s).hasDataType(DataTypes.STRING));
     }
 
+    @Test
+    public void test_resolve_eq_function_for_bit_types_with_different_length() throws IOException {
+        var e = SQLExecutor.builder(clusterService)
+            .addTable("create table tbl (b bit(3))")
+            .build();
+        var eq = (Function) e.asSymbol("tbl.b = B'111'");
```

Comment:
Shouldn't this test compare bitstrings with different lengths instead?
```suggestion
        var e = SQLExecutor.builder(clusterService).build();
        var eq = (Function) e.asSymbol("B'100' = B'10'");
```

Reply:
had to get back table statement so that expression gets resolved to a function, otherwise it's a Literal.